packages/serializers/md/src/remark-slate/ast-types.ts

```diff
@@ -0,0 +1,236 @@
+  heading: {
+    1: string;
+    2: string;
+    3: string;
+    4: string;
+    5: string;
+    6: string;
+  };
+  emphasis_mark: string;
+  strong_mark: string;
+  delete_mark: string;
+  inline_code_mark: string;
+  thematic_break: string;
+  image: string;
+}
+
+type RecursivePartial<T> = {
+  [P in keyof T]?: RecursivePartial<T[P]>;
+};
+
+export interface OptionType<T extends InputNodeTypes = InputNodeTypes> {
+  nodeTypes?: RecursivePartial<T>;
+  linkDestinationKey?: string;
+  imageSourceKey?: string;
+  imageCaptionKey?: string;
+}
+
+export interface MdastNode {
+  type?: MdastNodeType;
+  ordered?: boolean;
+  value?: string;
+  text?: string;
+  children?: Array<MdastNode>;
+  depth?: 1 | 2 | 3 | 4 | 5 | 6;
+  url?: string;
+  alt?: string;
+  lang?: string;
+  // mdast metadata
+  position?: any;
+  spread?: any;
+  checked?: any;
+  indent?: any;
+}
+
+export type TextNode = { text?: string | undefined };
+
+export type CodeBlockNode<T extends InputNodeTypes> = {
+  type: T['code_block'];
+  language: string | undefined;
+  children: Array<TextNode>;
+};
+
+export type HeadingNode<T extends InputNodeTypes> = {
+  type:
+    | T['heading'][1]
+    | T['heading'][2]
+    | T['heading'][3]
+    | T['heading'][4]
+    | T['heading'][5]
+    | T['heading'][6];
+  children: Array<DeserializedNode<T>>;
+};
+
+export type ListNode<T extends InputNodeTypes> = {
+  type: T['ol_list'] | T['ul_list'];
+  children: Array<DeserializedNode<T>>;
+};
+
+export type ListItemNode<T extends InputNodeTypes> = {
+  type: T['listItem'];
+  children: Array<DeserializedNode<T>>;
+};
+
+export type ParagraphNode<T extends InputNodeTypes> = {
+  type: T['paragraph'];
+  break?: true;
+  children: Array<DeserializedNode<T>>;
+};
+
+export type LinkNode<T extends InputNodeTypes> = {
+  type: T['link'];
+  children: Array<DeserializedNode<T>>;
+  [urlKey: string]: string | undefined | Array<DeserializedNode<T>>;
+};
+
+export type ImageNode<T extends InputNodeTypes> = {
+  type: T['image'];
+  children: Array<DeserializedNode<T>>;
+  [sourceOrCaptionKey: string]: string | undefined | Array<DeserializedNode<T>>;
+};
+
+export type BlockQuoteNode<T extends InputNodeTypes> = {
+  type: T['block_quote'];
+  children: Array<DeserializedNode<T>>;
+};
+
+export type InlineCodeMarkNode<T extends InputNodeTypes> = {
+  type: T['inline_code_mark'];
+  children: Array<TextNode>;
+  language: string | undefined;
+};
+
+export type ThematicBreakNode<T extends InputNodeTypes> = {
+  type: T['thematic_break'];
+  children: Array<DeserializedNode<T>>;
+};
+
+export type ItalicNode<T extends InputNodeTypes> = {
+  [K in T['emphasis_mark']]: true;
+} & {
+  children: TextNode;
+};
+
+export type BoldNode = {
+  bold: true;
+  children: TextNode;
+};
+
+export type StrikeThoughNode = {
+  strikeThrough: true;
+  children: TextNode;
+};
+
+export type InlineCodeNode = {
+  code: true;
+  text: string | undefined;
+};
+
+export type DeserializedNode<T extends InputNodeTypes> =
```

Comment:
Plate has a type for each element (e.g. `TLinkElement`), we could reuse those. So we'd need to add the element packages in the deps.

We don't have any for marks as we use https://github.com/udecode/plate/blob/919b51283b91846ea2fbf6ea37efc800c9ca940b/examples/src/typescript/plateTypes.ts#L95, we could define those in basic marks package.